teuthology/task/install/rpm.py

```diff
@@ -99,6 +101,23 @@ def _zypper_removerepo(remote, repo_list):
             'sudo', 'zypper', '-n', 'removerepo', repo['name'],
         ])
 
+
+def _downgrade_packages(ctx, remote, pkgs, pkg_version, config):
+    downgrade_pkgs = config.get('downgrade_packages', [])
+    if not downgrade_pkgs:
+        return pkgs
+    # assuming we are going to downgrade packages with the same version
+    first_pkg = downgrade_pkgs[0]
+    installed_version = packaging.get_package_version(remote, first_pkg)
```

Comment:
please see the comment. it's only for downgrade the packages of the same version.

Reply:
I don't understand, if the downgrade_pkgs is a list of packages supposed to be downgraded, why only the first packages from the list is checked for the version?

Reply:
Are all packages from pkgs supposed to be the same version?

Reply:
So this function is only for usage with ceph packages which have unique version convention/

Reply:
@kshtsk this fix is designed to address the needs of https://github.com/ceph/ceph/pull/25463 . where librados2 and librbd1 are of the *same* version. and they are supposed to be downgrade in the *single* yum command. otherwise, yum will refuse to downgrade the package if we just pass either of them to `yum downgrade` separately.

Reply:
> So this function is only for usage with ceph packages which have unique version convention/

or for any project which creates multiple packages.